ports/mimxrt10xx/common-hal/busio/UART.c

```diff
@@ -79,126 +81,180 @@ void common_hal_busio_uart_construct(busio_uart_obj_t *self,
         mp_float_t timeout, uint16_t receiver_buffer_size, byte* receiver_buffer,
         bool sigint_enabled) {
 
-    // TODO: Allow none rx or tx
-
-    bool have_tx = tx != NULL;
-    bool have_rx = rx != NULL;
-    if (!have_tx && !have_rx) {
-        mp_raise_ValueError(translate("tx and rx cannot both be None"));
-    }
-
     self->baudrate = baudrate;
     self->character_bits = bits;
     self->timeout_ms = timeout * 1000;
 
-    const uint32_t rx_count = sizeof(mcu_uart_rx_list) / sizeof(mcu_periph_obj_t);
-    const uint32_t tx_count = sizeof(mcu_uart_tx_list) / sizeof(mcu_periph_obj_t);
+    bool is_onedirection = false;
+    if (!rx != !tx) {
```

Comment:
Please rewrite this check. It's not clear how it works.

Reply:
It's an XOR, converts both to bool and compares to ensure they're different. Would a comment be OK or do you have another notation for xoring that you'd prefer?

Reply:
I think the implicit bool conversion makes it more confusing and a comment would help. I'd suggest:

```c
// We are transmitting one direction if one pin is NULL and the other isn't.
bool is_onedirection = (rx != NULL) != (tx != NULL);
```

Reply:
@tannewt yes that's much prettier, and good points about the implicit conversion. Nice solution!